inferno/net.py

```diff
@@ -413,3 +417,29 @@ def __init__(
     def get_loss(self, y_pred, y, train=False):
         y_pred_log = torch.log(y_pred)
         return self.criterion_(y_pred_log, y)
+
+    def predict(self, X):
+        return self.predict_proba(X).argmax(1)
+
+
+class NeuralNetRegressor(NeuralNet):
+    def __init__(
+            self,
+            module,
+            criterion=torch.nn.MSELoss,
+            *args,
+            **kwargs
+    ):
+        super(NeuralNetRegressor, self).__init__(
+            module,
+            criterion=criterion,
+            *args,
+            **kwargs
+        )
+
+    def check_data(self, _, y):
+        # The problem with 1-dim float y is that the pytorch DataLoader will
+        # somehow upcast it to DoubleTensor
+        if get_dim(y) == 1:
+            raise ValueError("The target data shouldn't be 1-dimensional; "
```

Comment:
But then the explanation above is not satisfactory, since it neither explains why DataLoader is suddenly relevant, nor why applying DataLoader would have this consequence. For the latter, we don't even know the explanation, so we would have to say: 'Because that's how it is'. Which is why we can as well save the user some time and directly put it as is.

In my opinion, an exception does not need to give a reason, it should say what is wrong and what can be done about it. Reasons should be left to code comments.